work_queue/src/network_function.py

```diff
@@ -50,18 +52,17 @@ def main():
             if event_size:
                 # receive the event itself
                 event = conn.recv(event_size)
-
                 event = json.loads(event)
-                print('event: {}'.format(event))
-
+                print('Network function: recieved event: {}'.format(event), file=sys.stderr)
                 # create a forked process for function handler
-                manager = multiprocessing.Manager()
-                response = manager.dict()
-                p = multiprocessing.Process(target=getattr(wq_worker_coprocess, function_name), args=(event, response))
-                p.start()
-                p.join()
-
-                response = json.dumps(dict(response))
+                p = os.fork()
+                if p == 0:
+                    response = getattr(wq_worker_coprocess, function_name)(event)
+                    os.write(write, json.dumps(response).encode("utf-8"))
+                    os._exit(-1)
+                response = os.read(read, 9999999999)
```

Comment:
Here let's put the os.read in a loop that reads 64K  (65536) at a time, as we do in link.c, until os.read returns the empty string.  Put each read result in a list an concatenate at the end.

Reply:
because we are reading from a pipe and not a file, if we try to read when there is nothing left in the pipe then python blocks instead of returning the empty string. As a result, I read chunks from the pipe until we got a chunk that was not 64k (indicating it has the last of the data)